src/rgw/rgw_rest_s3.cc

```diff
@@ -3283,6 +3283,8 @@ void RGWCopyObj_ObjStore_S3::send_partial_response(off_t ofs)
     set_req_state_err(s, op_ret);
     dump_errno(s);
 
+    dump_header_if_nonempty(s, "x-amz-copy-source-version-id", src_object->get_instance());
+    dump_header_if_nonempty(s, "x-amz-version-id", dest_object->get_instance());
```

Comment:
looks correct to me, but it's hard to tell whether `src_object` and `dest_object` are guaranteed to be initialized here. do we need to test them for null first?

cc @dang

Reply:
I think it's okay.  It's only called from send_response(), or ultimately from execute(), both of which should have src_object adn dest_object initialized.

Reply:
even though `op_ret` may be nonzero here? what if we got ENOENT looking up the src_object?

Reply:
Good point.  I missed that it's not bailing if op_ret is failure.  Maybe that is enough?  If not, then rgw::sal::RGWObject::empty() will need to be used to check that they're okay.

Reply:
@Rjerk can you please surround these two lines with an `if (op_ret == 0) {` block?

Reply:
done